var/spack/repos/builtin/packages/pythia8/package.py

```diff
@@ -131,16 +131,16 @@ def configure_args(self):
                 args.append("--with-boost=" + self.spec["boost"].prefix)
 
         if "+madgraph5amc" in self.spec:
-            args += "--with-mg5mes=" + self.spec["madgraph5amc"].prefix
+            args.append("--with-mg5mes=" + self.spec["madgraph5amc"].prefix)
         else:
-            args += "--without-mg5mes"
+            args.append("--without-mg5mes")
```

Comment:
I am not sure, whether this works, could you try and look into [`with_or_without`](https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html#with-or-without)?

Maybe something like this? UNTESTED!
```python
args.extend(self.with_or_without('mg5mes',
    variant='madgraph5amc',
    activation_value='prefix'))
```

Reply:
This is an `Autotools` package so it should work.  See docs at https://spack.readthedocs.io/en/latest/build_systems/autotoolspackage.html#activation-value.

Reply:
I don't think that will work for these packages because the name of the variant is different from the name of the package, unlike all the others `with_or_without` that are there in the recipe